File: src/main/kotlin/com/itmo/microservices/demo/payment/api/controller/PaymentController.java
package com.itmo.microservices.demo.payment.api.controller;

import com.itmo.microservices.demo.payment.api.model.PaymentSubmissionDto;
import com.itmo.microservices.demo.payment.api.model.UserAccountFinancialLogRecordDto;
import com.itmo.microservices.demo.payment.api.service.PaymentService;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RequiredArgsConstructor
@RequestMapping("/payment")
@RestController
public class PaymentController {

    private final PaymentService paymentService;

    @GetMapping("/finlog")
    List<UserAccountFinancialLogRecordDto> getFinlog(
            @AuthenticationPrincipal UserDetails user,
            @RequestParam(value = "order_id", required = false) UUID orderId
    ) {
        return paymentService.getFinlog(user.getUsername(), orderId);
    }

    @PostMapping
    PaymentSubmissionDto executeOrderPayment (
            @AuthenticationPrincipal UserDetails user,
            @RequestParam("order_id") UUID orderId
    ) {
        return paymentService.executeOrderPayment(user, orderId);
    }
}

File: src/main/kotlin/com/itmo/microservices/demo/payment/api/controller/PaymentController.kt
package com.itmo.microservices.demo.payment.api.controller

import com.itmo.microservices.demo.payment.api.model.PaymentSubmissionDto
import com.itmo.microservices.demo.payment.api.model.UserAccountFinancialLogRecordDto
import com.itmo.microservices.demo.payment.impl.service.PaymentServiceImpl
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestParam
import java.util.*

@Controller
class PaymentController(private val paymentServiceImpl: PaymentServiceImpl) {


    //Оплата заказа
    @PostMapping("/orders/{order_id}/payment")
    fun postPaymentSubmission(order_id: UUID): PaymentSubmissionDto {
        return paymentServiceImpl.orderPayment(order_id)
    }
    //Получение информации о финансовых операциях с аккаунтом пользователя
    @GetMapping("/finlog")
    fun getUserAccountFinInfoLog(@RequestParam order_id: UUID): List<UserAccountFinancialLogRecordDto> {
        return paymentServiceImpl.getPaymentFinLog(order_id)
    }

}